project/Build.scala

```diff
@@ -582,12 +579,23 @@ object Build {
       base = file("tools/js"),
       settings = myScalaJSSettings ++ commonToolsSettings ++ Seq(
           crossVersion := ScalaJSCrossVersion.binary,
-          resources in Test <+= jsTestDefinitions in (testSuite, Test),
-          jsDependencies += {
-            val artifact =
-              (artifactPath in (testSuite, Test, jsTestDefinitions)).value
-            ProvidedJS / artifact.getName % "test"
-          }
+          resourceGenerators in Test <+= Def.task {
+            val base = (resourceManaged in Compile).value
+            IO.createDirectory(base)
+            val outFile = base / "js-test-definitions.js"
+
+            val writer = new BufferedWriter(new FileWriter(outFile))
+            try {
+              org.scalajs.sbtplugin.HTMLRunnerTemplate.renderTestDefinitions(
```

Comment:
I guess this is the internal reason for which `HTMLRunnerTemplate` is public?
If that's the only reason, we can avoid it by making it `private[sbtplugin]`, and have a forwarder in `project/HTMLRunnerTemplateAccess.scala` like this:
```scala
package org.scalajs.sbtplugin

object HTMLRunnerTemplateAccess {
  def renderTestDefinitions(...): Unit =
    HTMLRunnerTemplate.renderTestDefinitions(...)
}
```
and use that from `Build.scala`.

Ugly, but less ugly than having to publicly expose something that should not be public.